packages/is-shallow-equal/src/objects.js

```diff
@@ -7,36 +7,24 @@
  * @return {boolean} Whether the two objects are shallow equal.
  */
 export default function isShallowEqualObjects( a, b ) {
-	if ( a === b ) {
-		return true;
-	}
-
 	const aKeys = Object.keys( a );
 	const bKeys = Object.keys( b );
 
 	if ( aKeys.length !== bKeys.length ) {
 		return false;
 	}
 
-	let i = 0;
-
-	while ( i < aKeys.length ) {
+	for ( let i = 0, len = aKeys.length; i < len; i++ ) {
 		const key = aKeys[ i ];
-		const aValue = a[ key ];
```

Comment:
```
let i = aKeys.length
while ( i-- ) {}
```
might be the fastest? I'm not sure, that's just what I always thought

Reply:
Because now you check `i < len` and increase the counter

Reply:
Are reverse loops with `--i` really faster? I'm already assigning `aKeys.length` to a variable, so I'm not evaluating `.length` on each iterations. I would expect the JS engine to reliably optimize the loop so that the details (`++i` vs `i++`, `++i` vs `--i`) don't matter.

Reply:
It's two operations instead of one: `i < len; i++` instead of `i--`